Add `start_prototype`, a tool used for writing entry point assembly.

jrvanwhy opened the pull request:
This is a Rust version of the entry point. Using it directly is undefined behavior for multiple reasons (Rust assumes the stack is setup and the globals are initialized), but it can be a useful reference while working on a real (assembly) entry point.

**Question for reviewers:** Do you think this belongs in the repository?

alistair23 commented:
Interesting. I don't think we should include code we aren't using that just confuses people (for example they edit it and re-build and nothing changes).

It does seem very useful though. Maybe it could be in documentation? The only problem then is that it could diverge.

I would say it doesn't belong here. If you wanted to you could copy the code into comments in the assembly file. That way it's all included in the same place.

jrvanwhy commented:
> I would say it doesn't belong here. If you wanted to you could copy the code into comments in the assembly file. That way it's all included in the same place.

How about `asm/start_prototype.rs`? The problem with putting into the assembly directly is the assembly files are architecture-specific while `start_prototype` is supported to work on all supported architectures (obviously with per-arch assembly to get `pc` and set `sp`).

alistair23 commented:
That could work.

Can we add a big comment at the top of the file saying this isn't compiled as well?

jrvanwhy commented:
> That could work.
> 
> Can we add a big comment at the top of the file saying this isn't compiled as well?

I've moved the file to `asm/` and updated the comment at the top indicating it is not compiled.

alistair23 commented:
bors r+
